quiche/src/h3/frame.rs

```diff
@@ -567,8 +585,11 @@ fn parse_settings_frame(
             0x0 | 0x2 | 0x3 | 0x4 | 0x5 =>
                 return Err(super::Error::SettingsError),
 
-            // Unknown Settings parameters must be ignored.
-            _ => (),
+            // Unknown Settings parameters go into additional_settings.
+            _ => match &mut additional_settings {
+                Some(s) => s.push((identifier, value)),
+                None => additional_settings = Some(vec![(identifier, value)]),
+            },
```

Comment:
Couple of points:

1. you can use get_or_insert to make the code cleaner like

```
_ => {
  let s: &mut Vec<(u64, u64)> = additional_settings.get_or_insert_default();
  s.push((identifier, value));
}
```

2. Is it really useful to capture grease settings types? I don't think we should waste effort trying to detect grease values but something to think about how an application will handle this.

Reply:
Concerning your point 2: 
The notion of "additional params" is specific to the sending-side.
On the receiving-side, the application retrieves the settings using the `.peer_settings_raw()` method.
I think that the behaviour on the receiving side is thus the same as before since the grease settings were already part of the "raw settings" retrieved by the application before.